cluster/src/main/scala/io/snappydata/impl/LeadImpl.scala

```diff
@@ -142,6 +142,10 @@ class LeadImpl extends ServerImpl with Lead
     bootProperties.setProperty(serverGroupsProp, groups)
     bootProperties.setProperty(STORE_PREFIX + Attribute.GFXD_HOST_DATA, "false")
     bootProperties.setProperty(STORE_PREFIX + Attribute.GFXD_PERSIST_DD, "false")
+    // default value for spark.sql.files.maxPartitionBytes in snappy is 32mb
+    if (bootProperties.getProperty("spark.sql.files.maxPartitionBytes") == null) {
+      bootProperties.setProperty("spark.sql.files.maxPartitionBytes", "33554432")
+    }
```

Comment:
It might be better to keep this in ServiceUtils.setCommonBootDefaults so that all such SD defaults are in one place.

Reply:
Actually had earlier added this in ServiceUtils.setCommonBootDefaults, but realized that the property wasn't taking effect on lead as only `bootProperties `are being set in Spark conf by a call to `conf.setAll(bootProperties.asScala)` in `LeadImpl#start` Let me know if I am missing something here.

Reply:
Oh ok. Then perhaps LeadImpl.start should have an explicit call to ServiceUtils.setCommonBootDefaults at the start like in ServerImpl.start. Otherwise the other Spark properties being set may not have been applying correctly either.

Reply:
Changed the code as per the review comment. Please have a look.